src/main/java/org/apache/sling/xss/impl/AntiSamyPolicyAdapter.java

```diff
@@ -0,0 +1,289 @@
+                        }
+                        else if (!patternList.isEmpty()) {
+                            policyBuilder.allowAttributes(attribute.getName())
+                                    .matching(matchesToPatterns(patternList))
+                                    .onElements(tag.getValue().getName());
+                        }
+                    }
+
+                    if (!styleSeen) {
+                        policyBuilder.allowAttributes(CssValidator.STYLE_ATTRIBUTE_NAME)
+                                .matching(cssValidator.newCssAttributePolicy()).onElements(tag.getValue().getName());
+                    }
+                    break;
+
+                default:
+                    throw new RuntimeException("No tag action found.");
+            }
+        }
+
+        // disallow style tag on specific elements
+        policyBuilder.disallowAttributes(CssValidator.STYLE_ATTRIBUTE_NAME)
+                .onElements(cssValidator.getDisallowedTagNames().toArray(new String[0]));
+
+        // ---------- dynamic attributes ------------
+        Map<String, Attribute> dynamicAttributes = new HashMap<>();
+
+        // checks if the dynamic attributes are allowed
+        if (policy.getDirectives().get(ALLOW_DYNAMIC_ATTRIBUTES).equals("true")) {
+            dynamicAttributes.putAll(policy.getDynamicAttributes());
+            for (Attribute attribute : dynamicAttributes.values()) {
+                if (attribute.getOnInvalid().equals(REMOVE_TAG_ON_INVALID_ACTION)) {
+                    onInvalidRemoveTagList.add(attribute.getName());
+                }
+
+                List<Pattern> regexsFromAttribute = attribute.getPatternList();
+                for (Pattern regex : regexsFromAttribute) {
+                    dynamicAttributesPolicyMap.put(attribute.getName(), newDynamicAttributePolicy(regex));
+                }
+
+                List<String> allowedValuesFromAttribute = attribute.getLiterals();
+                if (!allowedValuesFromAttribute.isEmpty()) {
+                    dynamicAttributesPolicyMap.put(attribute.getName(),
+                            newDynamicAttributePolicy(true, allowedValuesFromAttribute.toArray(new String[0])));
+                }
+
+            }
+        }
+
+        policyFactory = policyBuilder.allowTextIn(CssValidator.STYLE_TAG_NAME).toFactory();
+
+    }
+
+    public PolicyFactory getHtmlCleanerPolicyFactory() {
+        return policyFactory;
+    }
+
+    public Map<String, AttributePolicy> getDynamicAttributesPolicyMap() {
+        return dynamicAttributesPolicyMap;
+    }
+
+    public List<String> getOnInvalidRemoveTagList() {
+        return onInvalidRemoveTagList;
+    }
+
+    public CssValidator getCssValidator() {
+        return cssValidator;
+    }
+
+    private static Predicate<String> matchesToPatterns(List<Pattern> patternList) {
+        return new Predicate<String>() {
+            @Override
+            public boolean apply(String s) {
```

Comment:
Rather use https://docs.oracle.com/javase/8/docs/api/java/util/regex/Pattern.html#asPredicate-- together with https://docs.oracle.com/javase/8/docs/api/java/util/function/Predicate.html#or-java.util.function.Predicate-

Reply:
This would cause type issues. 
The html sanitizer uses the `com.google.common.base.Predicate` and the proposed methods are from `java.util.function`.

Reply:
You can convert from one to the other via https://guava.dev/releases/22.0/api/docs/com/google/common/base/Predicate.html#test-T-

Reply:
Right, but I'm not sure what that would get us. We need to supply a Guava Predicate to https://javadoc.io/doc/com.googlecode.owasp-java-html-sanitizer/owasp-java-html-sanitizer/latest/org/owasp/html/HtmlPolicyBuilder.AttributeBuilder.html